pkg/views/user/create/view.go

```diff
@@ -67,3 +72,18 @@ func CreateUserView(createView *CreateView) {
 		log.Fatal(err)
 	}
 }
+
+
+func IsValidPassword(password string) bool {
+	if !regexp.MustCompile(`^.{8,128}$`).MatchString(password) {
```

Comment:
This can be enhanced by employing a single regex.

Reply:
Go supports RE2 syntax, so lookaround assertions is not support in RE2. We can use another packages for checking validation.
[https://support.google.com/docs/thread/215946555/data-validation-for-password-must-have-at-least-one-upper-case-character?hl=en](https://support.google.com/docs/thread/215946555/data-validation-for-password-must-have-at-least-one-upper-case-character?hl=en)

Reply:
https://github.com/goharbor/harbor-cli/blob/a0a9b98b8a5e04c7560f1a6710b437679f42d8cd/pkg/utils/utils.go#L62

You can use the above method instead of relying on regexp.

### Since, considering performance of both methods:
- **Character-by-Character Validation:** This method might be slightly more **performant** because it only iterates through the string once and uses direct character checks.

- **Regular Expressions:** This method involves multiple regex compilations and matches. Each MatchString call processes the string separately, which can be less efficient.

> Note: go regexp doesn't support backtracking.